projects/core/src/main/resources/data/computercraft/lua/rom/programs/shell.lua

```diff
@@ -87,7 +148,7 @@ function shell.execute(command, ...)
         local sDir = fs.getDir(sPath)
         local env = createShellEnv(sDir)
```

Comment:
I think we need to push this into `executeProgram` too, and use the resolved path of the hashbang program/interpreter instead of the original file. The problem here is that CC's `require` implementation resolves modules relative to the program's directory (rather than PUC Lua, which uses the current directory), so currently the hashbang program will use the wrong program path!

This might be a little cleaner if we do my above suggestion with passing arguments (so we'd have `executeProgram(100, sPath, { [0] = command, ... })` here).